src/controller/controller.cpp

```diff
@@ -327,25 +323,20 @@ void Controller::onDialogOK(const CardCertificateAndPinInfo& cardCertAndPinInfo)
 void Controller::onDialogCancel()
 {
     qDebug() << "User cancelled";
-
     disposeUI();
-
-    writeResponseToStdOut(isInStdinMode,
-                          makeErrorObject(RESP_USER_CANCEL, QStringLiteral("User cancelled")),
-                          commandType());
+    _result = makeErrorObject(RESP_USER_CANCEL, QStringLiteral("User cancelled"));
+    writeResponseToStdOut(isInStdinMode, _result, commandType());
     exit();
 }
 
 void Controller::onCriticalFailure(const QString& error)
 {
     qCritical() << "Exiting due to command" << std::string(commandType())
                 << "fatal error:" << error;
-    writeResponseToStdOut(isInStdinMode, makeErrorObject(RESP_TECH_ERROR, error), commandType());
```

Comment:
My thinking was that we should write the response before disposing the UI to assure that the response does not get lost in case dispose exits with error. The response is more important than disposal. What do you think about this?